src/module/item/condition/persistent-damage-editor.ts

```diff
@@ -84,8 +98,45 @@ class PersistentDamageDialog extends appv1.api.Application<PersistentDamageDialo
         return input.reportValidity();
     }
 
-    override activateListeners($html: JQuery<HTMLElement>): void {
-        const html = $html[0];
+    static #onAdd(this: PersistentDamageEditor, event: PointerEvent) {
+        const section = htmlClosest(event.target, ".persistent-entry");
+        if (!section) return;
+
+        const elements = this.#getInputElements(section);
+        const formula = elements.formula?.value.trim() || "1d6";
+        const damageType = elements.damageType?.value;
+        const dc = Number(elements.dc?.value) || 15;
+
+        if (this.#reportFormulaValidity(`(${formula})[${damageType}]`, elements.formula)) {
+            const baseConditionSource = game.pf2e.ConditionManager.getCondition("persistent-damage").toObject();
+            const persistentSource = fu.mergeObject(baseConditionSource, {
+                system: {
+                    persistent: { formula, damageType, dc },
+                },
+            });
+            this.actor.createEmbeddedDocuments("Item", [persistentSource]);
+        }
+    }
+
+    static #onDelete(this: PersistentDamageEditor, event: PointerEvent) {
+        const existingId = htmlClosest(event.target, ".persistent-entry[data-id]")?.dataset.id;
+        const existing = this.actor.items.get(existingId, { strict: true });
+        existing.delete();
+    }
+
+    static #onRoll(this: PersistentDamageEditor) {
+        const existing = this.actor.itemTypes.condition.filter((c) => c.slug === "persistent-damage");
+        for (const condition of existing) {
+            condition.onEndTurn();
+        }
+    }
+
+    protected override async _onRender(
```

Comment:
Use `_onChangeForm` here

Reply:
I will need to research this, but its not a form, the fields don't have names, so I'll have to see how well it works. This will take a bit before I can say yes/no for real.

Reply:
Looks like it worked, though its a bit ugly since its not a real form.